jekyll/_cci2/api-developers-guide.md

```diff
@@ -0,0 +1,177 @@
+---
+layout: classic-docs
+title: "CircleCI API Developer's Guide"
+short-title: "Developer's Guide"
+description: "API cookbook for internal and external CircleCI developers"
+categories: [getting-started]
+order: 1
+---
+
+This *API Developer's Guide* was written to assist developers in quickly and easily making API calls to CircleCI services to return detailed information about users, pipelines, projects and workflows.
+
+* TOC
+{:toc}
+
+# API Overview
+
+The CircleCI platform provides a powerful API that enables users to retrieve detailed information about users, jobs, workflows and pipelines.
+
+**Note:** Whenever you see a term prefixed with a colon (:), please note that this value must be replaced by a user-input value. For example `:token` in the above call must be replaced by the token string generated in your account:
+
+curl https://circleci.com/api/v1.1/me?circle-token=dkanmbowk34iajdkl3iakdthisisarandomtoken
+
+## API Classes
+
+The current classes of the API v2 endpoints are:
+
+- Authentication
+- Pipeline
+- Workflows
+- User (Preview)
+- Project (Preview)
+- Job (Preview)
+
+**Note:** Portions of the CircleCI API v2 remain under “Preview”. Preview endpoints are not yet fully supported or considered generally available. Breaking changes to API v2 Preview endpoints are planned in advance and are announced in the API v2 breaking changes log.
+
+## Authentication and Authorization
+
+The CircleCI API utilizes token-based authentication to manage access to the API server and validate that a user has permission to make API requests. Before you can make an API request, you must first add an API token and then verify that you are authenticated by the API server to make requests. The process to add an API token and have the API server authenticate you is described in the sections below.
+
+**Note** You may use the API token as the username for HTTP Basic Authentication, by passing the `-u` flag to the `curl` command.
+
+### Add an API Token
+
+To add an API token, perform the steps listed below.
+
+1.  Add an API token from your [account dashboard](https://circleci.com/account/api).
+2.  To test it, [view it in your browser](https://circleci.com/api/v1.1/me) or call the API using the command below.
+
+```sh
+$ curl https://circleci.com/api/v1.1/me?circle-token=:token
```

Comment:
As above, maybe we can replace this with a v2 example? - just because that's the focus of the doc

Reply:
Fixed